content/en/blog/2022-03-09-pod-security-standard-restricted.md

```diff
@@ -0,0 +1,151 @@
+> - the `seccomp` profile was set to the runtime default
+> - run as non-root was enabled
+> - the filesystem group was set to 1337
+> - the user and group ID was set to 65534
+>
+> Flux also enables the Seccomp runtime default across all controllers.
+> Why is this important? Well, the default `seccomp` profile blocks key
+> system calls that can be used maliciously, for example to break out of
+> the container isolation. The recently disclosed [kernel vulnerability
+> CVE-2022-0185](https://blog.aquasec.com/cve-2022-0185-linux-kernel-container-escape-in-kubernetes)
+> is a good example of that.
+
+## Pod Security Standards definition
+
+Kubernetes defined three policies in its Pod Security Standards. They
+range from
+
+- **Privileged**: This does not place any restrictions on the workload
+  at all. The idea being that this can be used for system- and
+  infrastructure-level workloads which are managed by privileged and
+  trusted users only.
+- **Baseline**: This policy comes with some restrictions. It aims to
+  guard against known privilege escalations while still making it
+  easy to adopt and use it by keeping a certain level of
+  compatibility with most workloads.
+- **Restricted**: Inheriting all the restrictions from *Baseline*, it
+  enforces additional limitations, thus follows hardening best
+  practices by increasing the isolation levels the workload is
+  exposed to.
+
+We are very pleased that all Flux controllers were moved to
+*Restricted*, as that offers the highest level of security for you.
+
+We recommend checking out the [Upstream Kubernetes documentation on Pod
+Security
+Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)
+as it gives a generally good overview of all the security features
+enabled. In addition to that you can see which restrictions were added
+as part of which Kubernetes release, meaning that with every Kubernetes
+release, you will benefit from new Upstream Kubernetes security
+improvements automatically.
+
+{{% note %}}
+As of v1.24 Kubernetes still runs all workloads with `seccomp` in
+`unconfined` mode, in other words, disabled. On the other hand, Docker
+has `seccomp` enabled by default for years now.
+
+There are discussions to change the Kubernetes default on v1.25, and have all
+workloads set to `RuntimeDefault` unless opted-out. This would be based on
+`SeccompDefault` [feature
+gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+being enabled from that version onwards.
+{{% /note %}}
+
+{{% note %}}
+If you are an OpenShift user, you might run into [this
+issue](https://github.com/fluxcd/source-controller/issues/582)
+([related upstream
+report](https://github.com/openshift/cluster-kube-apiserver-operator/issues/1325)).
+The work-around right now is to remove the seccomp profile as
+described in [these instructions](/docs/use-cases/openshift/#flux-installation-with-cli).
+{{% /note %}}
+
```

Comment:
@pjbgf WDYT?

Reply:
Yes, @dholbach I like it - precise and to the point, whilst providing all the info the user may need. :+1: